### PR TITLE
UG-956: Add showMessageOverlay function to manageArticleLists component

### PR DESCRIPTION
**Pull request by meel-io**
### Description

- Adds the `MessageElement` component that includes the markup of the fake door message and, as an argument, expects a click handler for its button.
- Adds the `showMessageOverlay` method to display the fake door message for this smoke test. This method passes the a handler to the `MessageElement` with logic to close the messageOverlay and restore the createListOverlay after clicking on "Continue"
- Adds logic in the `createList` callback to close the `createListOverlay` and show the messageOverlay if the **Public** toggle is checked.

### How to test this code?

- Check out this branch and run `npm link`
- Clone `next-article` and  run `npm link @financial-times/n-myft-ui`
- Run `next-article` with `npm start`
- Switch on the `manageArticleLists` and `myftListPublicPrivateToggle` flags on [Toggler](https://toggler.ft.com/#manageArticleLists)
- Click on "Save" and the `manageArticleLists` modal should appear.
- Click on "Add to a new list" to display the form. Create a new list making sure the Public toggle is checked
- You should see the fake door message informing you that this feature is not available (See screenshots). Click on continue and the `manageArticleLists` modal should reappear.

| Desktop |Mobile |
| ------ | ----- |
|Screenshot 2022-08-15 at 16 04 00|Screenshot 2022-08-15 at 16 03 38|

Please make sure the changes are not impacting these consumer apps in unexpected ways:
 
- [x] [next-myft-page](https://github.com/Financial-Times/next-myft-page) ([Live](https://www.ft.com/myft), [Local](https://local.ft.com:5050/myft))
- [x] [next-video-page](https://github.com/Financial-Times/next-video-page) ([Live](https://www.ft.com/video), [Local](https://local.ft.com:5050/video))
- [x] [next-tour-page](https://github.com/Financial-Times/next-tour-page) ([Live](https://www.ft.com/tour), [Local](https://local.ft.com:5050/tour))
- [x] [next-search-page](https://github.com/Financial-Times/next-search-page) ([Live](https://www.ft.com/search), [Local](https://local.ft.com:5050/search))

### CODE REVIEW

**Review by jamesnicholls — CHANGES_REQUESTED** [review]
Notification works great. I don't think it was introduced in this PR, but I'm seeing this visual bug on screen sizes (~394px wide):

Screenshot 2022-08-16 at 12 53 51

File: myft/ui/save-article-to-list-variant.js [hunk 1]
```diff
@@ -302,6 +347,26 @@ function ListCheckboxElement (addToList, removeFromList) {
 	};
 }
 
+function MessageElement (onContinue) {
+	const message = `
+	<div class="myft-ui-create-list-variant-message-content">
```
[the remark this answers is not in the document]
Comment (meel-io) [reply myft/ui/save-article-to-list-variant.js]: Thanks for keeping an eye on accessibility. Setting it to `alert` didn't work well with the created list announcement. Using the `aria-live=polite` does the trick: It announces the created list and then it reads the content of the fake door message

File: myft/ui/save-article-to-list-variant.js [hunk 2]
```diff
@@ -195,8 +234,14 @@ function FormElement (createList, showPublicToggle) {
 			triggerCreateListEvent(contentId, listId);
 			triggerAddToListEvent(contentId, listId);
 			positionOverlay(createListOverlay.wrapper);
+
+			if (inputIsShareable.checked) {
```
[the remark this answers is not in the document]
Comment (meel-io) [reply myft/ui/save-article-to-list-variant.js]: Good point 👍 I've added this change

File: myft/ui/save-article-to-list-variant.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jamesnicholls) [reply myft/ui/save-article-to-list-variant.js]: This works great. In fact, the whole component is really nice to use with a screen reader.

**Review by jamesnicholls — APPROVED** [review]
The notification works great. I'm still getting the visual bug described above, but happy to address this in a different PR.

**Pull request merged**